### PR TITLE
Fix ReduceMean + GlobalAveragePool

### PR DESCRIPTION
**Pull request by BraunPhilipp**
* `matchAndReplace` with appropriate `aten.mean.dim` op

### CODE REVIEW

File: src/Conversion/ONNXToTorch/NN/ReduceMean.cpp [hunk 1]
```diff
@@ -60,27 +60,30 @@ struct ONNXReduceMeanOpToTorchLowering : public ConversionPattern {
...
-    // **TODO**: reduceMean.axes is not being used currently. Probably use
-    // AtenMeanDimOp instead of AtenMean, or use both.
-
     mlir::MLIRContext *context = reduceMean.getContext();
     Location loc = reduceMean.getLoc();
 
     auto axis = mlir::extractFromI64ArrayAttr(reduceMean.axesAttr());
     if(!(axis.size() == 2 && axis[0] == 2 && axis[1] == 3))
-      op->emitError("Not implemented yet for general axis sizes");
+      assert("Not implemented yet for general axis sizes");
```
Comment (ljfitz) [inline src/Conversion/ONNXToTorch/NN/ReduceMean.cpp:68]: Does `assert` result in a more graceful failure than `op->emitError`? I had expected the opposite.

Of note also is that an assert is for sanity checking, as in you do not expect it to be possible/legal to create the condition that is being asserted, which is why optimized builds can delete asserts. If legal onnx source could have different values here, and we want to fail to convert .

File: src/Conversion/ONNXToTorch/NN/ReduceMean.cpp [hunk 2]
```diff
@@ -60,27 +60,30 @@ struct ONNXReduceMeanOpToTorchLowering : public ConversionPattern {
...
-                           ? rewriter.create<ConstantIntOp>(loc, keepDims)
-                           : getIntValue(0, rewriter, context, loc);
 
     auto dataTensor = getTorchTensor(reduceMean.data(), rewriter, context, loc);
     auto resultType = toTorchType(context, reduceMean.getResult().getType());
 
-    Value result =
-        rewriter.create<AtenMeanOp>(loc, resultType, dataTensor, keepdimVal);
+
+    auto sintType = IntegerType::get(getContext(), 64, IntegerType::SignednessSemantics::Signed);
```
Comment (ljfitz) [inline src/Conversion/ONNXToTorch/NN/ReduceMean.cpp:78]: Is `rewriter.getIntegerType(unsigned width, bool isSigned)` available?

**Review by ljfitz — APPROVED** [review]
Thanks Philipp. My only concern here is the use of `assert`, but I suspect this may be an issue to address in more places as I suspect you're harmonizing practices, and we should switch those practices to not assert but properly fail on valid but unsupported inputs.

File: src/Conversion/ONNXToTorch/NN/ReduceMean.cpp (hunk 1 above)
Comment (BraunPhilipp) [reply src/Conversion/ONNXToTorch/NN/ReduceMean.cpp:68]: Just had a read through other conversion passes. Standard practice seems to be `op->emitError` and failure in this case.

**Pull request merged**